Apps/phfireeyeetp/fireeyeetp_connector.py

```diff
@@ -0,0 +1,1491 @@
+                ), resp_json
+            )
+
+        return self._process_response(r, action_result)
+
+    def _save_file_to_vault(self, data, filename, container_id, action_result):
+        # Creating temporary directory and file
+        try:
+            if hasattr(Vault, 'get_vault_tmp_dir'):
+                temp_dir = Vault.get_vault_tmp_dir()
+            else:
+                temp_dir = "/opt/phantom/vault/tmp/"
+
+            temp_dir = "{}{}".format(temp_dir, '/{}'.format(uuid.uuid4()))
+
+            os.makedirs(temp_dir)
+
+            file_path = os.path.join(temp_dir, filename)
+
+            with open(file_path, 'wb') as file_obj:
+                file_obj.write(data)
+        except Exception as e:
+            err = self._get_error_message_from_exception(e)
+            return action_result.set_status(phantom.APP_ERROR, "Error while writing to temporary file", err), None
+
+        # Adding pcap to vault
+        vault_ret_dict = Vault.add_attachment(file_path, container_id, filename)
+
+        # Removing temporary directory created to download file
+        try:
+            os.rmdir(temp_dir)
+        except Exception:
+            return action_result.set_status(phantom.APP_ERROR, "Unable to remove temporary directory"), None
+
+        # Updating data with vault details
+        if vault_ret_dict['succeeded']:
+            vault_details = {
+                phantom.APP_JSON_VAULT_ID: vault_ret_dict[phantom.APP_JSON_HASH],
+                'file_name': filename
+            }
+            return phantom.APP_SUCCESS, vault_details
+
+        # Error while adding report to vault
+        self.debug_print('Error adding file to vault:', vault_ret_dict)
+        action_result.append_to_message('. {}'.format(vault_ret_dict['message']))
+
+        # Set the action_result status to error, the handler function will most probably return as is
+        return phantom.APP_ERROR, None
+
+    def _paginator(self, endpoint, action_result, data, method="get", **kwargs):
+        """ This function is used to handle the gathering of alerts for the list alerts action.
+        :param endpoint: API endpoint to use to get the alerts
+        :param action results: Action results for Phantom
+        :param data: dict of parameters to send to the API endpoint
+        :param method: HTTP method to use when calling the API endpoint
+        :param **kwargs: Optional and additional arguments to use for calling the API endpoint. Note: these parameters need to be valid Python Requests parameters
+        :return: a list of alerts
+        """
+        items_list = list()
+
+        while True:
+            ret_val, items = self._make_rest_call(endpoint, action_result, json=data, method=method, **kwargs)
+
+            if phantom.is_fail(ret_val):
+                return action_result.get_status(), None
+            else:
+                limit = items.get('meta').get('total') - len(items_list)
```

Comment:
these chained `.get` calls without a default seem a little odd since if any of the `.get` calls before the last one doesn't find the key, it will end up raising an exception for doing a `None.get(...)`. on this particular line even moreso since the subtraction will fail with a `None`

can we just make these direct retrievals e.g. `items['meta']['total']` to indicate they're expected to exist (unlike the conditional existence cases like on line 356)? and should we also have a `try` around these to gracefully handle failures?

this comment applies to both this function `_paginator` and the one after `_paginator2` which are pretty similar but there seem to be other functions that do the same e.g. `_create_container`

Reply:
Fixed